### ISSUE TITLE
Inline functions

### ISSUE THREAD

**Issue by jorolf** [thebennybox-Community/Community-Compiler#48]
Inline functions are like normal functions except that they replace the call to the function with the function content. Imagine this:

```
inline infix fn add(a: i32, b: i32): i32 {
    a + b
} 

fn test() {

```

**Comment by davidgarland** [comment]
Imo we ougt to implement it via pragmas or attributes rather than polluting the namespace with keywords that could be expressed as attributes easily. For example C++ uses [[]], Nim uses {..}, etc.

**Comment by jorolf** [comment]
we already have #30 for attribute discussion.

Another thing this would enable us to do is replace the operators with infix functions that are replaced by their actual operators at compile time

**Comment by davidgarland** [comment]
@jorolf yea I know, im saying adding more keywords to the language that arent control flow or typing related is kinda pointless *because* we *already have* attributes, not that we *should* add attributes.

**Comment by MyvarHD** [comment]
adding more key words is not a flexibel system, becuase you have to add the keyword in the lexer and parser thats why attributes make more sense, plus adding the ability to access meta information form atributes is a very usefull things, as for why when with the @ symobol (args) syntax is that simply put it makes for an easy way to make the attribute meta info avalible at runtime, becuase it is literly a function invoke prefixed by an @, we can eather then somehow use prefix to invoke the atibutes at runtime and extract the data that way or have the compiler turn them into invoekes and the the invokes at the start of the execution that way we can easly integrate things for runtime access

**Comment by MyvarHD** [comment]
no, there parser uses aneneric syntax for attributes the semantics engine and code generator is where you would have to add the attribute
